drivers/pds/lib/utils.go

```diff
@@ -1074,6 +1076,131 @@ func RunTpccWorkload(dbUser string, pdsPassword string, dnsEndpoint string, dbNa
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{"app": benchmarkName},
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name:            "consul-agent",
+							Image:           consulAgentImage,
+							ImagePullPolicy: corev1.PullAlways,
+							Env: []corev1.EnvVar{
+								{
+									Name: "AGENT_TOKEN",
+									ValueFrom: &corev1.EnvVarSource{
+										SecretKeyRef: &corev1.SecretKeySelector{
+											LocalObjectReference: corev1.LocalObjectReference{
+												Name: deploymentName + "-creds",
+											},
+											Key: "master_token",
+										},
+									},
+								},
+								{
+									Name:  "PDS_CLUSTER",
+									Value: deploymentName,
+								},
+								{
+									Name:  "PDS_NS",
+									Value: namespace,
+								},
+							},
+						},
+						{
+							Name:            "consul-kv-workload",
+							Image:           consulAgentImage,
+							ImagePullPolicy: corev1.PullAlways,
+							Command:         []string{"/bin/kv-workload.sh"},
+							Env: []corev1.EnvVar{
+								{
+									Name: "AGENT_TOKEN",
+									ValueFrom: &corev1.EnvVarSource{
+										SecretKeyRef: &corev1.SecretKeySelector{
+											LocalObjectReference: corev1.LocalObjectReference{
+												Name: deploymentName + "-creds",
+											},
+											Key: "master_token",
+										},
+									},
+								},
+							},
+						},
+						{
+							Name:            "consul-bench",
+							Image:           consulBenchImage,
+							ImagePullPolicy: corev1.PullAlways,
+							Env: []corev1.EnvVar{
+								{
+									Name: "CONSUL_HTTP_TOKEN",
+									ValueFrom: &corev1.EnvVarSource{
+										SecretKeyRef: &corev1.SecretKeySelector{
+											LocalObjectReference: corev1.LocalObjectReference{
+												Name: deploymentName + "-creds",
+											},
+											Key: "master_token",
+										},
+									},
+								},
+								{
+									Name: "SERVICE_NAME",
+									ValueFrom: &corev1.EnvVarSource{
+										FieldRef: &corev1.ObjectFieldSelector{
+											FieldPath: "metadata.name",
+										},
+									},
+								},
+								{
+									Name:  "SERVICE_INSTANCES",
+									Value: "10",
+								},
+								{
+									Name:  "SERVICE_FLAP_SECONDS",
+									Value: "10",
+								},
+								{
+									Name:  "SERVICE_WATCHERS",
+									Value: "10",
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+	log.InfoD("Going to Trigger Consul Bench Workload for the Deployment")
+	deployment, err := k8sApps.CreateDeployment(deploymentSpec, metav1.CreateOptions{})
+	if err != nil {
+		log.Errorf("An Error Occured while creating deployment %v", err)
+		return nil, err
+	}
+	err = k8sApps.ValidateDeployment(deployment, timeOut, timeInterval)
+	if err != nil {
+		log.Errorf("An Error Occured while validating the pod %v", err)
+		return nil, err
+	}
+
+	time.Sleep(1 * time.Minute)
```

Comment:
any specific reason for this wait

Reply:
Here we are simply running this workload for 1 minute

Reply:
can you please add a comment, the reason for waiting

Reply:
Done